typegraph/python/typegraph/runtimes/kv.py

```diff
@@ -0,0 +1,85 @@
+# Copyright Metatype OÜ, licensed under the Mozilla Public License Version 2.0.
+# SPDX-License-Identifier: MPL-2.0
+
+from dataclasses import dataclass
+from typing import Optional
+
+from typegraph import fx, t
+from typegraph.gen.exports.runtimes import (
+    BaseMaterializer,
+    Effect,
+    KvMaterializer,
+    KvRuntimeData,
+)
+from typegraph.gen.types import Err
+from typegraph.runtimes.base import Materializer, Runtime
+from typegraph.wit import runtimes, store
+
+
+class KvRuntime(Runtime):
+    host: str
+    port: Optional[str]
+    db_number: Optional[int]
+    password: Optional[str]
+
+    def __init__(
+        self,
+        host: str,
+        port: Optional[str],
+        db_number: Optional[int],
+        password: Optional[str],
+    ):
+        data = KvRuntimeData(
+            host=host, port=port, db_number=db_number, password=password
+        )
+        runtime_id = runtimes.register_kv_runtime(store, data)
+        if isinstance(runtime_id, Err):
+            raise Exception(runtime_id.value)
+
+        super().__init__(runtime_id.value)
+        self.host = host
+        self.port = port
+        self.db_number = db_number
+        self.password = password
+
+    def get(self):
+        mat = self.__operation(KvMaterializer.GET, fx.read())
+        return t.func(t.struct({"key": t.string()}), t.string(), mat)
+
+    def set(self):
+        mat = self.__operation(KvMaterializer.SET, fx.update())
+        return t.func(
+            t.struct({"key": t.string(), "value": t.string()}), t.string(), mat
+        )
```

Comment:
How about making it generic `def set(self, value: typedef)` ?
On the kv runtime (gate), the value would be json stringified for example.

Reply:
@afmika Good point, also `def get(self, value: typedef)`. But we will also need an encoding field ("raw" or "json" or other...).
Later, we could also add prefix.

Reply:
Is it better to just live it like this or...